splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/ScanOperation.java

```diff
@@ -450,4 +472,40 @@ public int getSplits() {
     public FormatableBitSet getAccessedColumns() throws StandardException{
         return scanInformation.getAccessedColumns();
     }
+
+    private long mapToTxId(DataValueDescriptor dataValue) throws StandardException {
+        if(dataValue instanceof SQLTimestamp) {
+            Timestamp ts = ((SQLTimestamp)dataValue).getTimestamp(null);
+            SpliceLogUtils.trace(LOG,"time travel ts=%s", ts.toString());
+            try {
+                return SIDriver.driver().getTxnStore().getTxnAt(ts.getTime());
+            } catch (IOException e) {
+                throw Exceptions.parseException(e);
+            }
+        }else if(dataValue instanceof SQLTinyint || dataValue instanceof SQLSmallint || dataValue instanceof SQLInteger || dataValue instanceof SQLLongint) {
+            return dataValue.getLong();
+        }else {
+            throw StandardException.newException(SQLState.NOT_IMPLEMENTED); // fix me, we should read SqlTime as well.
```

Comment:
Given the code for `SQLTimestamp` above, it seems supporting `SQLTime` is trivial. Any reason to leave it for now?

Reply:
It was not required so far and I am not exactly sure about its semantics since `SQLTime` is a time period ranging from 00:00:00.0000000 to 23:59:59.9999999 I think we need to map it to a point of time correctly (to act similar to `Timestamp`, i.e. let's say the current date is `01.01.2020` and given `SQLTime` is `01:02:03.456789`, the resulting `Timestamp` would be `01.01.2020 01:02:03.456789`. I think `SQLTime#getTimestamp(Calendar)` provides this functionality already.
But to answer your question, I will add support for this data type later since it wasn't required so far IIRC.